payments-core/src/main/java/com/stripe/android/paymentsheet/injection/PaymentSheetViewModelModule.kt

```diff
@@ -59,81 +52,47 @@ internal class PaymentSheetViewModelModule {
         stripeAccount = lazyPaymentConfiguration.get().stripeAccountId
     )
 
-    @Provides
-    @Singleton
-    fun provideStripeApiRepository(
-        application: Application,
-        lazyPaymentConfiguration: Lazy<PaymentConfiguration>
-    ) = StripeApiRepository(
-        application,
-        { lazyPaymentConfiguration.get().publishableKey }
-    )
-
     @Provides
     @Singleton
     fun provideStripeIntentRepository(
         stripeApiRepository: StripeApiRepository,
-        lazyPaymentConfig: Lazy<PaymentConfiguration>
+        lazyPaymentConfig: Lazy<PaymentConfiguration>,
+        @IOContext workContext: CoroutineContext
     ): StripeIntentRepository {
         return StripeIntentRepository.Api(
             stripeRepository = stripeApiRepository,
             requestOptions = ApiRequest.Options(
                 lazyPaymentConfig.get().publishableKey,
                 lazyPaymentConfig.get().stripeAccountId
             ),
-            workContext = Dispatchers.IO
+            workContext = workContext
         )
     }
 
     @Provides
     @Singleton
     fun providePaymentMethodsApiRepository(
         stripeApiRepository: StripeApiRepository,
-        lazyPaymentConfig: Lazy<PaymentConfiguration>
+        lazyPaymentConfig: Lazy<PaymentConfiguration>,
+        @IOContext workContext: CoroutineContext
     ): PaymentMethodsRepository {
         return PaymentMethodsApiRepository(
             stripeRepository = stripeApiRepository,
             publishableKey = lazyPaymentConfig.get().publishableKey,
             stripeAccountId = lazyPaymentConfig.get().stripeAccountId,
-            workContext = Dispatchers.IO
+            workContext = workContext
         )
     }
 
-    @Provides
-    @Singleton
-    fun providePaymentFlowResultProcessor(
-        application: Application,
-        starterArgs: PaymentSheetContract.Args,
-        lazyPaymentConfig: Lazy<PaymentConfiguration>,
-        stripeApiRepository: StripeApiRepository,
-    ): PaymentFlowResultProcessor<out StripeIntent, StripeIntentResult<StripeIntent>> {
-        return when (starterArgs.clientSecret) {
-            is PaymentIntentClientSecret -> PaymentIntentFlowResultProcessor(
-                application,
-                { lazyPaymentConfig.get().publishableKey },
-                stripeApiRepository,
-                enableLogging = true,
-                Dispatchers.IO
-            )
-            is SetupIntentClientSecret -> SetupIntentFlowResultProcessor(
-                application,
-                { lazyPaymentConfig.get().publishableKey },
-                stripeApiRepository,
-                enableLogging = true,
-                Dispatchers.IO
-            )
-        }
-    }
-
     @Provides
     @Singleton
     fun provideGooglePayRepository(
-        application: Application,
+        appContext: Context,
```

Comment:
Can you annotate with `@ApplicationContext`?

Reply:
What prompted the need for changing from `Application` to `Context`?

Reply:
Re: `@ApplicationContext` - we'll need to create the annotation manually, at the moment there is no dependencies actually needs it, so will just keep it as is.

Re: change from `Application` to `Context` - `Application` was [passed into](https://github.com/stripe/stripe-android/blob/0b3bfb9d08e6e6a02ca609b2323d3c81db3eb86b/payments-core/src/main/java/com/stripe/android/paymentsheet/injection/PaymentSheetViewModelComponent.kt#L19) `PaymentSheetViewModelComponent` because it is required to be passed to its [super constructor](https://github.com/stripe/stripe-android/blob/0b3bfb9d08e6e6a02ca609b2323d3c81db3eb86b/payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt#L86).  Since it's also a `Context` we just directly used it to initialize all dependencies in this Module including `PaymentFlowResultProcessor` and `PaymentController`.  However they were created with `Context` from `FlowControllerModule`, in order to merge them into one shared `Module`, we need to make the interface the same